include/fe/fe.h

```diff
@@ -793,6 +793,13 @@ class FE : public FEGenericBase<typename FEOutputType<T>::type>
                                       std::vector<Number> & nodal_soln_on_side,
                                       bool add_p_level = true);
 
+  /**
+   * Whether we cache the node locations on the last
+   * element we computed on to try to avoid calling
+   * init_shape_functions and compute_shape_functions
+   */
+  bool caching;
```

Comment:
This should probably be a static member - even if anyone is doing something as weird as turning caching on for some FE objects and off for others, that means they're deep enough in the weeds that they could figure out how to do that by manipulating the static member.

Reply:
I've removed this member entirely. I can put it back if you think we should?